ennMethods.rb

```diff
@@ -0,0 +1,101 @@
+# Module
+module Enumerable
+  #my_each: clone of each
+  def my_each
+    for x in self do
+      yield(x)
+    end
+    return self
+  end
+
+  #my_each_with_index: clone of each.with_index
+  def my_each_with_index
+    for i in 0...self.length do
+      yield(self[i], i)
+    end
+    return self
+  end
+
+  #my_select: clone of select
+  def my_select
+    i = 0
+    result = []
+    self.my_each do |x|
+      if yield(x)
+        result[i] = x
+        i += 1
+      end
+    end
+    return result
+  end
+
+  #my_all?: clone of all?
+  def my_all?
+    test = true
+    self.my_each do |x|
+      test = test && yield(x)
+    end
+    return test
+  end
+
+  #my_any?: clone of any?
+  def my_any?
+    test = false
+    self.my_each do |x|
+      test = test || yield(x)
+    end
+    return test
+  end
+
+  #my_none?: clone of none?
+  def my_none?
+    test = true
+    self.my_each do |x|
+      test = test || !yield(x)
+    end
+    return test
+  end
```

Comment:
Not working
```Ruby
> %w{ant bear cat}.none? { |word| word.length >= 4 }
 => false 
> %w{ant bear cat}.my_none? { |word| word.length >= 4 }
 => true 
> [nil, false, true].none?
 => false 
> [nil, false, true].my_none?
 => true 
```

Reply:
Change made.
Fixed the logic of `my_none?`.
It now accepts a pattern as optional parameter and it also accepts that the user inputs no block.